### ISSUE TITLE
libgpuarray pip install with Anaconda can't find headers

### ISSUE THREAD

**Issue by jessejohns** [Theano/libgpuarray#265]
Hello,

When I am installing pygpu with Anaconda and using pip install, I have to specify the include directory in **init**.py as follows:

```
def get_include():
    import os.path
    p = '/home/jesse/Programs/anaconda/pygpu'
    assert os.path.exists(os.path.join(p, 'gpuarray_api.h'))
    return p
```

Installing with python setup.py install, everything works out well.

**Comment by abergeron** [comment]
Is the path you are specifying the one to your copy of the source?  In that case the problem is that some versions of pip/setuptools/distutils decide that a .h is not part of the package data (even though I mark it as such) and don't install it.

There are also some versions of Debian which decide to move it in a random directory that doesn't make sense.

I have not been able to figure out why it does that yet, although I haven't been spending too much time on it either since I can't reproduce on any of my machines.  I remember someone writing about MANIFEST.in which could help so you can try that.

**Comment by jessejohns** [comment]
Hi there,

Pip ends up creating its down directory and throws the *.h into anaconda/pygpu and the .py into /anaconda/lib/python2.7/site-packages/pygpu.  I'm using the python/pip from the local anaconda installation and using ubuntu 16.04.  I'm able to reproduce the behavior on RHEL6 as well.

**Comment by abergeron** [comment]
If you could find out why it does that and how to fix it that would help.  The .h file should be alongside the .py file.

Alternatively if there is a way to retrieve the path where pip want to hide files (without hardcoding) then that could be an alternative.

**Comment by astooke** [comment]
Hey I think I just ran into a similar problem, possibly the same.  After doing all the bleeding edge Theano + libgpuarray + pygpu installation instructions (into a conda env), and updating library paths with libgpuarray install location (not the same as where I cloned to), running Theano with gpu would fail like so:

```
Problem occurred during compilation with the command line below:
...
fatal error: gpuarray/array.h: No such file or directory
compilation terminated.
```

Solution for me was less elegant: manually put a folder `gpuarray` with all the header files from `libgpuarray\include` in my anaconda env site-package: `/opt/anaconda2/envs/py3/lib/python3.5/site-packages/pygpu-0.6.0rc1-py3.5-linux-x86_64.egg/pygpu` (this is one of the places shown as always included, -I, in the Theano compilation attempt in the error message).

Maybe I missed a step?

**Comment by abergeron** [comment]
The libgpuarray headers have to be installed somewhere where the compiler can find them, yes.  Normally a regular install will place them in /usr/local/include, but if you put them elsewhere you have to make the compiler aware of that.

**Comment by nouiz** [comment]
Just a note, conda package are in preparation, so this should be eaiser
when they are ready.

**Comment by abergeron** [comment]
This should be fixed in v0.6.0.  Please reopen if you still have issues.

**Comment by astooke** [comment]
In a different conda environment, I just did `conda install -c rdonnelly theano=0.9.0beta1`, which automatically included libgpuarray-0.6.0 and pygpu-0.6.0, and had the same problem.  I don't see a different channel for theano-0.9.0beta1 or pygpu-0.6.0...?

**Comment by abergeron** [comment]
You have to use the git master of Theano for it to work.  0.9 beta1 was still broken in that regard.
